straxen/plugins/acqmon_processing.py

```diff
@@ -0,0 +1,130 @@
+    """
+    __version__ ='0.0.3'
+        
+    depends_on = ('raw_records_aqmon') 
+    provides  = ('aqmon_hits')
+    data_kind = ('aqmon_hits')
+    
+    dtype = strax.hit_dtype
+
+    save_when = strax.SaveWhen.TARGET
+
+    def compute(self, raw_records_aqmon):
+        rec = strax.raw_to_records(raw_records_aqmon)
+        strax.zero_out_of_bounds(rec)
+        strax.baseline(rec, baseline_samples = self.config['baseline_samples_aqmon'], flip = True)
+        aqmon_hits = strax.find_hits(rec, min_amplitude = self.config['hit_min_amplitude_aqmon'])
+    
+        return aqmon_hits 
+
+    
+
+#### Veto hardware ####:
+# V1495 busy veto module: 
+# Generates a 25 ns NIM pulse whenever a veto begins and a 25 ns NIM signal when it ends. 
+# A new start signal can occur only after the previous busy instance ended.
+# 1ms (1e6 ns) - minimum busy veto length, or until the board clears its memory
+    
+# DDC10 High Energy Veto:
+# 10ms (1e7 ns) - fixed HE veto length in XENON1T DDC10, 
+# in XENONnT it will be calibrated based on the length of large S2 SE tails
+# The start/stop signals for the HEV are generated by the V1495 board
+  
+@export
+@strax.takes_config(
+    strax.Option('max_veto_gap', default = int(5e8),
+                 help = 'Maximum separation between veto stop and start pulses [ns]'),
+    strax.Option('channel_map', track = False, type = immutabledict, 
+                 help = 'immutabledict mapping subdetector to (min, max)'
+                      'channel number.'))
+
+class VetoIntervals(strax.OverlapWindowPlugin):
+    """ Find pairs of veto start and veto stop signals and the veto duration between them
+    busy_*  <= V1495 busy veto for tpc channels
+    he_*    <= V1495 busy veto for high energy tpc channels
+    hev_*   <= DDC10 hardware high energy veto
+    """
+        
+    __version__ ='0.0.24'
+    depends_on = ('aqmon_hits') 
+    provides  = ('veto_intervals')
+    data_kind = ('veto_intervals')
+
+    def infer_dtype(self):
+        dtype = [(('veto start time since unix epoch [ns]', 'time'), np.int64),
+                 (('veto end time since unix epoch [ns]', 'endtime'), np.int64),
+                 (('veto interval [ns]', 'veto_interval'), np.int64),
+                 (('veto signal type', 'veto_type'), np.str('U10'))]
+        return dtype
+    
+    def setup(self):
+        aq_ch_start = straxen.n_hard_aqmon_start
+        aqmon_channel_names = ('sum_wf','m_veto_sync',
+                               'hev_stop', 'hev_start', 'he_stop','he_start','busy_stop', 'busy_start')
+        channel_numbers = np.arange(aq_ch_start, aq_ch_start + len(aqmon_channel_names))
+        
+        # Keeping ch_range as class attribute for now, maybe will need it for other features
+        self.channel_range = self.config['channel_map']['aqmon']
+        self.channel_map = dict(zip(aqmon_channel_names, channel_numbers))
+        self.veto_names = ['busy_','he_','hev_']
+        return self.channel_map
```

Comment:
@jorana is the setup return used by anything in strax? I do not think so.

Reply:
Sharp. I don't recall any instance of such a usage of return

Reply:
I think I've seen it somewhere but can't remember where. I can remove it, though it seems to be harmless, or I'm missing something?

Reply:
Yes, you can just remove it

Reply:
Why does it matter though?